### PR TITLE
Delete test images test-by-test for multiprocessing concerns

### PR DESCRIPTION
**Pull request by AlanCoding**
still no idea if this will work

### CODE REVIEW

**Comment by AlanCoding** [comment]
Nope, doesn't work.

**Comment by AlanCoding** [comment]
hold on... that might just have passed integration

**Review by beeankha — APPROVED** [review]

**Comment by AlanCoding** [comment]
yay

```
github-workflows : SUCCESS in 1m 22s
ansible-tox-linters : SUCCESS in 4m 51s
ansible-tox-py36 : SUCCESS in 4m 57s
ansible-tox-py37 : SUCCESS in 6m 26s
ansible-build-python-tarball : SUCCESS in 3m 51s
ansible-builder-tox-integration : SUCCESS in 50m 41s
```

File: .zuul.d/project.yaml [hunk 1]
```diff
@@ -2,5 +2,7 @@
 - project:
     check:
       jobs:
-        - ansible-builder-tox-integration:
-            voting: false
+        - ansible-builder-tox-integration
+    gate:
+      jobs:
+        - ansible-builder-tox-integration
```
Comment (pabelanger) [inline .zuul.d/project.yaml:8]: this is correct, thanks!

**Comment by pabelanger** [comment]
FAILED test/integration/test_build.py::test_build_fail_exitcode[podman] looks to be failing

**Comment by AlanCoding** [comment]
That's what we saw in the prior check run, so this shows that it's not consistent.

It's almost certainly a bug.

> error while running runtime: exit status 127

https://github.com/ansible/ansible-builder/blob/2118e25a340338175c6f74ecb11b50ef94a0e2ed/ansible_builder/utils.py#L86

So `127` is not `None`... and `rc` is not 0, because ansible-builder thought that the command was successful. That means that it must have been `None`.

https://stackoverflow.com/questions/2995983/using-subprocess-wait-and-poll

> poll always returns immediately. It effectively does a wait with a timeout of 0, catches any exception, and returns None if the process hasn't completed

Best guess is that the process hasn't completed, and this is a bug in our `run_command` method.

**Comment by AlanCoding** [comment]
Here's a reproducer:

```python
>>> [run_command(['sleep', '1', '|', 'flase']) for i in range(20)]
An error occured (rc=1), see output line(s) above for details.
An error occured (rc=1), see output line(s) above for details.
An error occured (rc=1), see output line(s) above for details.
An error occured (rc=1), see output line(s) above for details.
An error occured (rc=1), see output line(s) above for details.
An error occured (rc=1), see output line(s) above for details.
[(None, []), (None, []), (None, []), (None, []), (None, []), (None, []), (None, []), (None, []), (None, []), (None, []), (None, []), (1, []), (None, []), (None, []), (1, []), (1, []), (1, []), (None, []), (1, []), (1, [])]
```

That doesn't really leave anything to the imagination about this flaky behavior.

**Comment by AlanCoding** [comment]
That wasn't testing what I thought it was:

```
$ python -c "from ansible_builder.utils import run_command; print(run_command(['sleep', '1', '&&', 'false'], allow_error=True, capture_output=True))"
(1, ['sleep: invalid time interval ‘&&’', 'sleep: invalid time interval ‘false’', "Try 'sleep --help' for more information."]
```

Just wanted to disclose that. Now I'll figure out another version that does what I want and put it in a real test.

File: ansible_builder/utils.py [hunk 2]
```diff
@@ -82,7 +82,7 @@ def run_command(command, capture_output=False, allow_error=False):
         logger.debug(line.rstrip('\n'))  # line ends added by logger itself
     logger.debug('')
 
-    rc = process.poll()
+    rc = process.wait()
```
Comment (AlanCoding) [inline ansible_builder/utils.py:85]: @shanemcd this here is important and I want to get you to weigh in.

This fixes the `test_failed_command` tests being added in this PR, for what it's worth.

Comment (shanemcd) [reply ansible_builder/utils.py:85]: Wow! Incredible. 💯

**Pull request merged**